lesson-7/script.js

```diff
@@ -0,0 +1,157 @@
+*/
+const createNewTeam = (team_name, driver_name, driver_birth_year, car_mark, team_sponsor, driver_admission) => {
+    if (isString(team_name) || !isString(driver_name) || !isNumber(driver_birth_year) || !isString(car_mark) || !isString(team_sponsor) || !isBoolean(driver_admission)) {
+        return 'Some arguments has an incorrect type.'
+    } else if (driver_birth_year >= 2004 && driver_birth_year <= 1962) {
+        return 'It seems that driver year of birth is not real or allowed.'
+    }
+    return {
+        team_name,
+        driver_name,
+        driver_birth_year,
+        car_mark,
+        team_sponsor,
+        driver_admission,
+    }
+}
+const team_1 = createNewTeam ('1', 'ros', 1994, 'true', 'ford', true);
+/* 
+2. Запитати дані (через prompt та циклічно) про нового користувача у системі та зберегти його в об'єкт. 
+Дані про користувача - логін, пароль, місто, країна, стать, вік
+*/
+let newUserTemplate = {
+    login: "empty",
+    password: "empty",
+    city: "empty",
+    country: "empty",
+    gender: "empty",
+    age: "empty",
+}
+const createUser = () => {
+    let newUser = {}
+    for (let key in newUserTemplate) {
+        let collectUserData = prompt(`Enter your ${key}, please.`);
+        if (key === 'login' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'password' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'city' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'country' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'gender' && (!isString(collectUserData) || isNumber(Number(collectUserData)))
+            || key === 'age' && !isNumber(Number(collectUserData))) {
+            return console.log('Some arguments has an incorrect type.');
+        } else if (key === 'age') {
+            newUser[key] = Number(collectUserData);    
+        } else {
+            newUser[key] = collectUserData;
+        }
+    }
+    return newUser
+}
+const newUser_1 = createUser()
+/* 3. Створити функцію, яка буде міняти дані в конкретного користувача створеного пунктом вище. 
+Наприклад сhangeUserData(user_1, city, 'Kherson). 
+Де user_1 - обʼєкт в якому буде мінятись, city - поле, яке буде мінятися на нове значення - 'Kherson'
+*/
+let askNewCity = prompt("Enter your new city.")
+const changeUserData = (user_num, key, value) => {
+    if (key in user_num) {
+        if (key === 'login' && (!isString(value) || isNumber(Number(value)))
+            || key === 'password' && (!isString(value) || isNumber(Number(value)))
+            || key === 'city' && (!isString(value) || isNumber(Number(value)))
+            || key === 'country' && (!isString(value) || isNumber(Number(value)))
+            || key === 'gender' && (!isString(value) || isNumber(Number(value)))
```

Comment:
You have already function that checks for `Number`. Better to add that there and use it as

```suggestion
            || key === 'gender' && (!isString(value) || isNumber(value))
```

Reply:
A lot of repeated code